static/js/portico/signup.js

```diff
@@ -15,9 +15,19 @@ $(function () {
         };
     }
 
-    $('#registration, #password_reset').validate({
+    $('#id_password').on('change keyup', function () {
+        var acceptable = password_quality($(this).val(), $('#pw_strength .bar'));
+
+        if (acceptable) {
+            $('#id_password')[0].setCustomValidity('');
+        } else {
+            $('#id_password')[0].setCustomValidity('Password is weak.');
```

Comment:
I think it's better to have a if-else block then have a ternary operator as a function argument.